agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -98,38 +196,80 @@ func getTaskFromRequest(state dockerstate.TaskEngineState, r *http.Request) (*ap
+		return nil, http.StatusBadRequest, errors.New("invalid request: no task was found")
 	}
 
 	task, found := state.TaskByArn(taskARN)
 	if !found {
 		logger.Critical("No task was found for taskARN for task protection request", logger.Fields{
 			loggerfield.TaskARN: taskARN,
 		})
-		return nil, http.StatusInternalServerError, errors.New("Failed to find a task for the request")
+		return nil, http.StatusInternalServerError, errors.New("failed to find a task for the request")
 	}
 
 	return task, http.StatusOK, nil
 }
 
 // GetTaskProtectionHandler returns a handler function for GetTaskProtection API
-func GetTaskProtectionHandler(state dockerstate.TaskEngineState,
-	cluster string) func(http.ResponseWriter, *http.Request) {
+func GetTaskProtectionHandler(state dockerstate.TaskEngineState, credentialsManager credentials.Manager,
+	factory TaskProtectionClientFactoryInterface, cluster string) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		getTaskProtectionRequestType := "api/v1/GetTaskProtection"
+		getTaskProtectionRequestType := "api/GetTaskProtection/v1"
 
 		task, responseCode, err := getTaskFromRequest(state, r)
 		if err != nil {
 			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
 			return
 		}
 
-		// TODO: Call ECS
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
 		logger.Info("GetTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster: cluster,
 			loggerfield.TaskARN: task.Arn,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", getTaskProtectionRequestType)
+
+		response, err := ecsClient.GetTaskProtection(&ecs.GetTaskProtectionInput{
+			Cluster: aws.String(cluster),
+			Tasks:   aws.StringSlice([]string{task.Arn}),
+		})
+
+		if err != nil {
+			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
+			logger.Info(fmt.Sprintf("ECS throws an exception with type %s", exceptionType))
+			writeJSONResponse(w, statusCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
+		logger.Debug("getTaskProtection response:", logger.Fields{
+			loggerfield.TaskProtection: response.ProtectedTasks,
+			loggerfield.Reason:         response.Failures,
+		})
+
+		// there are no exceptions but there are failures when getting protection in scheduler
+		if len(response.Failures) > 0 {
+			writeJSONResponse(w, http.StatusBadRequest, response.Failures[0], getTaskProtectionRequestType)
+			return
+		}
+
+		/**
+		 * handling different response scenarios
+		 * if protection is enabled, response contains a protectedTask with ProtectionEnabled value true
+		 * if protection expires for requested task, response contains a protectedTask with ProtectionEnabled value false
+		 * if protection is never set or specifically unset by user, response will be empty
```

Comment:
Note: Checked with Scheduler folks - this behavior will be changed, to assure there are consistent responses among these two scenarios. Expect code changes after that is ready from control plane.